### PR TITLE
[MISC] Secondary units stuck on configuring exporter

### PR DESCRIPTION
**Pull request by dragomirp**
## Issue
Starting the Postgresql exporter is frequently failing on replica units, likely due to LXD update.

## Solution
Trigger the exporter start in on peer data change, after Patroni and PG are started.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -137,10 +137,10 @@ jobs:
             echo "mark_expression=" >> $GITHUB_OUTPUT
           else
             echo Skipping unstable tests
-            echo "mark_expression=not unstable" >> $GITHUB_OUTPUT
+            echo "mark_expression=and not unstable" >> $GITHUB_OUTPUT
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:140]: This seems to have broken the scheduled tests.

File: src/backups.py [hunk 2]
```diff
@@ -411,6 +411,9 @@ def _initialise_stanza(self) -> None:
 
     @property
     def _is_primary_pgbackrest_service_running(self) -> bool:
+        if not self.charm.primary_endpoint:
+            logger.warning("Failed to contact pgBackRest TLS server: no promary endpoint")
+            return False
```
Comment (dragomirp) [inline src/backups.py:416]: [Sometimes](https://github.com/canonical/postgresql-operator/actions/runs/6157787902/job/16709350681) there's an unhandled exception due to subprocess None parameter.

File: src/grafana_dashboards/postgresql-metrics.json
Comment (dragomirp) [inline src/grafana_dashboards/postgresql-metrics.json:1]: To bring in line with DPE-2588

File: src/charm.py [hunk 3]
```diff
@@ -947,14 +950,6 @@ def _on_start(self, event: StartEvent) -> None:
 
         self.unit.set_workload_version(self._patroni.get_postgresql_version())
 
-        try:
-            # Set up the postgresql_exporter options.
-            self._setup_exporter()
-        except snap.SnapError:
-            logger.error("failed to set up postgresql_exporter options")
-            self.unit.status = BlockedStatus("failed to set up postgresql_exporter options")
```
Comment (dragomirp) [inline src/charm.py:955]: There will no longer be a blocked status over this, but the call in `_on_peer_relation_changed()` will retry if not successful.

**Review by marceloneppel — APPROVED** [review]
Thanks!

File: src/backups.py (hunk 2 above)
Comment (taurus-forever) [reply src/backups.py:416]: `s/promary/primary/`?

**Review by taurus-forever — APPROVED** [review]
LGTM, but one typo in text message.

**Pull request merged**